README.md

```diff
@@ -1,84 +1,275 @@
+});
+
+const program = new ProgramCall('QWCRSVAL', { lib: 'QSYS' });
+const outBuf = [
+  [0, '10i0'],
+  [0, '10i0'],
+  ['', '36h'],
+  ['', '10A'],
+  ['', '1A'],
+  ['', '1A'],
+  [0, '10i0'],
+  [0, '10i0'],
+];
+const errno = [
+  [0, '10i0'],
+  [0, '10i0', { setlen: 'rec2' }],
+  ['', '7A'],
+  ['', '1A'],
+];
+
+program.addParam(outBuf, { io: 'out' });
+program.addParam(66, '10i0');
+program.addParam(1, '10i0');
+program.addParam('QCCSID', '10A');
+program.addParam(errno, { io: 'both', len: 'rec2' });
+
+conn.add(program);
+
+
+conn.run((error, xmlOutput) => {
+  if (error) {
+    throw error;
+  }
+  const result = xmlToJson(xmlOutput);
+  console.log(result);
+});
+```
+### CommandCall
+CommandCall is used to execute a CL, QSH, or PASE command.
+
+#### Example
+```javascript
+const {
+  Connection, CommandCall, xmlToJson,
+} = require('itoolkit');
+
+const conn = new Connection({
+  transport: 'ssh',
+  transportOptions: { host: 'myhost', username: 'myuser', password: 'mypassword' }
+});
+
+conn.add(new CommandCall({ command: 'RTVJOBA USRLIBL(?) SYSLIBL(?)', type: 'cl' }));
+
+conn.run((error, xmlOutput) => {
+  if (error) {
+    throw error;
+  }
+  const result = xmlToJson(xmlOutput);
+  console.log(result);
+});
+```
+
+### SqlCall
+SqlCall is used to make an SQL query.
+
+#### Example
+
+```javascript
+const {
+  Connection, SqlCall, xmlToJson,
+} = require('itoolkit');
+
+const conn = new Connection({
+  transport: 'ssh',
+  transportOptions: { host: 'myhost', username: 'myuser', password: 'mypassword' }
+});
+
+const sql = new SqlCall();
+sql.prepare('call qsys2.tcpip_info()');
+sql.execute();
+sql.fetch();
+sql.free();
+
+conn.add(sql);
+
+conn.run((error, xmlOutput) => {
+  if (error) {
+    throw error;
+  }
+  const result = xmlToJson(xmlOutput);
+  console.log(result);
+});
+```
+## Utility Toolkit Functions
+
+Aside from the main classes this toolkit also provides helper functions to access:
+- Data Queues
+- User Space objects
+- Object info
+- Product info
+- Network info
+
+```js
+const {
+  Connection, Toolkit, xmlToJson,
+} = require('itoolkit');
+
+const conn = new Connection({
+  transport: 'ssh',
+  transportOptions: { host: 'myhost', username: 'myuser', password: 'mypassword' }
+});
+
+const toolkit = new Toolkit(conn);
+
+toolkit.getSysValue('QCCSID', (error, value) => {
+  if (error) {
+    throw error;
+  }
+  console.log(`QCCSID = ${value}`);
+});
+```
+
+# Documentation
+TODO port and update Docs from [developer works](https://www.ibm.com/developerworks/community/wikis/home?lang=en#!/wiki/IBM%20i%20Technology%20Updates/page/Toolkit%20for%20i%20APIs)
+
+# Testing
+Refer to the [README](test/README.md)
+
+# Contributing
+Please read the [contribution guidelines](https://github.com/IBM/nodejs-itoolkit/blob/master/CONTRIBUTING.md).
 
 # License
-[`MIT`](https://github.com/IBM/nodejs-itoolkit/blob/master/LICENSE) file.
+[`MIT`](https://github.com/IBM/nodejs-itoolkit/blob/master/LICENSE)
```

Comment:
There's a massive amount of changes to the README that have nothing to do with the SSH transport. Seems like this should have been done under a different PR.

Reply:
Yeah I do have unrelated changes in there in regard to ssh tranport. 

Should I backtrack those changes?

Reply:
I will backtrack changes to the readme